### ISSUE TITLE
How to add an imgCompressor?

### ISSUE THREAD

**Issue by Antiokus314** [nodeca/mincer#181]
Adding a jsCompressor/cssCompressor was fairly straightforward - since they were already extending onto the Mincer/Environment objects. However, add a custom image compression class is definitely not the case. 

#### Here's what I did
* Registered the mime type
* Created a custom class for image minification
* Followed the layout in compressing.js --- 124-152
```javascript
Mincer.registerMimeType('image/png', '.png')

var CustomImageCompressor = function CustomImageCompressor() {
  Mincer.Template.apply(this, arguments);
};

CustomImageCompressor.prototype.evaluate = function() {
  console.log('Custom image compressor');
// My image compression stuff;
};

Object.defineProperty(module.exports, 'imgCompressor', {
  get: function () {
    return this.__imgCompressor__ || null;
  },

  set: function (compressor) {
    var mime = 'image/png';

    if (this.__imgCompressor__) {
      this.unregisterBundleProcessor(mime, this.__imgCompressor__);
    }

    if (_.isFunction(compressor)) {
      this.__imgCompressor__ = compressor;
      this.registerBundleProcessor(mime, 'img_compressor', compressor);
      return;
    }

    if (this.__compressors__.get(mime)[compressor]) {
      this.__imgCompressor__ = this.__compressors__.get(mime)[compressor];
      this.registerBundleProcessor(mime, this.__imgCompressor__);
      return;
    }

    throw new Error('Unknown compressor: ' + compressor);
  }
});

Mincer.registerCompressor('image/png', 'myImgCompressor', CustomImageCompressor);
```

I then try to set it on my new environment object like so
```
var environment = new Mincer.Environment();
environment.imgCompressor = 'myImgCompressor'
```

However my ```console.log``` statement inside the ```evaluate``` method never gets called, so I know that it isn't working. Any guidance as to what I'm missing?

**Comment by ixti** [comment]
You don't need to define `imgCompressor` or such. `jsCompressor` and `cssCompressor` are commonly used post-processors, so they have such "helpers". For your local use case all you need to do is to register preProcessor or postProcessor (I would go with preProcessor):

``` javascript
var optim = require("optim");

Mincer.registerPreProcessor("image/png", "PngOptimizer", function (context) {
  this.data = optim(this.data);
});
```

Or, you can define a processor class like so:

``` javascript
var optim = require("optim");

function PngOptimizer() {
  Mincer.Template.apply(this, arguments);
}

require('util').inherits(PngOptimizer, Mincer.Template);

PngOptimizer.prototype.evaluate = function (/*context, locals*/) {
  this.data = optim(this.data);
};

// finally register your processor

Mincer.registerPreProcessor("image/png", PngOptimizer);
```

**Comment by Antiokus314** [comment]
Does that actually work? I tried implementing it and I found that the images were being mangled by Mincer. I made sure it wasn't the image compression library, and just made it load only the files at compile time (without doing anything)

e.g.
```javascript
var environment = new Mincer.Environment();
environment.appendPath(scripts_dir);
environment.appendPath(styles_dir);
environment.appendPath(images_dir);

var manifest = new Mincer.Manifest(environment, './public/assets');
manifest.compile(['**/*.(js|css|png)']);
```
The js and css files get copied over without being modified, however that doesn't seem to be the case with the images.

Is Mincer doing something to the image files during the compiling process that is making un-processable?
